File: Java/opentraceability/src/main/java/opentraceability/models/events/SensorElement.java
package opentraceability.models.events;

import java.util.List;
import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityObjectAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionElementsAttribute;

public class SensorElement {
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = "", name ="sensorMetadata", sequenceOrder = 0)
    public SensorMetaData metaData;

    @OpenTraceabilityArrayAttribute(itemType = SensorReport.class)
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = "", name = "sensorReport", sequenceOrder = 1)
    public List<SensorReport> reports;

    @OpenTraceabilityExtensionElementsAttribute
    public List<IEventKDE> extensionKDEs;
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/events/sensorelement.kt
package opentraceability.models.events

import opentraceability.interfaces.IEventKDE
import opentraceability.utility.attributes.*
import java.util.*

class SensorElement {

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute("","sensorMetadata", 0)
    var metaData: SensorMetaData? = null

    @OpenTraceabilityArrayAttribute
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute("","sensorReport", 1)
    var reports: MutableList<SensorReport> = mutableListOf()

    @OpenTraceabilityExtensionElementsAttribute
    var extensionKDEs: MutableList<IEventKDE> = mutableListOf()
}
